src/compiler/abstract_compiler.nit

```diff
@@ -93,6 +96,10 @@ redef class ToolContext
 			exit(1)
 		end
 
+		if opt_debug.value and opt_compile_dir.value == null then
```

Comment:
I force the generation of the source directory, since I find it convenient to have sources available around to be checking, though this might be me only (that and it's actually faster to write `nitc -g prgm.nit` instead of `nitc -g --compile-dir nit_compile prgm.nit`)